File: GrafoPorMatrixAdjacente/Java/Grafo.java
import java.util.ArrayList;
import java.util.List;

public class Grafo{
    private List<Integer> vertices;
    private int[][] matrizAdjacencia;

    public Grafo() {
        this.vertices = new ArrayList<>();
        this.matrizAdjacencia = new int[20][20];
    }

    public void adicionarVertice(int vertice) {
        if (vertice >= 0 && vertices.size() < 20) {
            vertices.add(vertice);
        } else {
            System.out.println("Vértice inválido ou limite máximo atingido");
        }
    }

    public void adicionarAresta(int verticeOrigem, int verticeDestino, int peso) {
        if (vertices.contains(verticeOrigem) && vertices.contains(verticeDestino)) {
            int indiceOrigem = vertices.indexOf(verticeOrigem);
            int indiceDestino = vertices.indexOf(verticeDestino);
            matrizAdjacencia[indiceOrigem][indiceDestino] = peso;
            matrizAdjacencia[indiceDestino][indiceOrigem] = peso;
        } else {
            System.out.println("Vértice de origem ou destino inválido");
        }
    }

    public void visualizarGrafo() {
        System.out.println("Matriz de Adjacência:");
        for (int i = 0; i < vertices.size(); i++) {
            for (int j = 0; j < vertices.size(); j++) {
                System.out.print(matrizAdjacencia[i][j] + " ");
            }
            System.out.println();
        }
    }

}

File: GrafoPorMatrixAdjacente/Rust/Grafo/src/grafo.rs
use std::collections::HashMap;

pub struct Grafo {
    vertices: Vec<u32>,
    arestas: HashMap<(u32, u32), u32>,
    max_vertices: usize,
}

impl Grafo {
    pub fn new(max_vertices: usize) -> Grafo {
        Grafo {
            vertices: Vec::new(),
            arestas: HashMap::new(),
            max_vertices,
        }
    }

    pub fn adicionar_vertice(&mut self, vertice: u32) {
        if vertice >= 0 && self.vertices.len() < self.max_vertices {
            self.vertices.push(vertice);
        } else {
            println!("Vértice inválido ou limite máximo atingido");
        }
    }

    pub fn adicionar_aresta(&mut self, vertice_origem: u32, vertice_destino: u32, peso: u32) {
        if self.vertices.contains(&vertice_origem) && self.vertices.contains(&vertice_destino) {
            self.arestas.insert((vertice_origem, vertice_destino), peso);
            self.arestas.insert((vertice_destino, vertice_origem), peso);
        } else {
            println!("Vértice de origem ou destino inválido");
        }
    }

    pub fn visualizar_grafo(&self) {
        println!("Matriz de Adjacência:");
        for &origem in &self.vertices {
            for &destino in &self.vertices {
                if let Some(&peso) = self.arestas.get(&(origem, destino)) {
                    print!("{:<2} ", peso);
                } else {
                    print!("0  ");
                }
            }
            println!();
        }
    }
}